cluster.go

```diff
@@ -1741,3 +1741,42 @@ func diffPeers(peers1, peers2 []peer.ID) (added, removed []peer.ID) {
 	}
 	return
 }
+
+// RepoGC performs garbage collection sweep on all peers' IPFS repo.
+func (c *Cluster) RepoGC(ctx context.Context) (*api.GlobalRepoGC, error) {
+	_, span := trace.StartSpan(ctx, "cluster/RepoGC")
+	defer span.End()
+	ctx = trace.NewContext(c.ctx, span)
+
+	members, err := c.consensus.Peers(ctx)
+	if err != nil {
+		logger.Error(err)
+		return nil, err
+	}
+	lenMembers := len(members)
+
+	ctxs, cancels := rpcutil.CtxsWithCancel(ctx, lenMembers)
+	defer rpcutil.MultiCancel(cancels)
+
+	repoGCsResp := make([]*api.RepoGC, lenMembers, lenMembers)
+
+	errs := c.rpcClient.MultiCall(
+		ctxs,
+		members,
+		"IPFSConnector",
+		"RepoGC",
+		struct{}{},
+		rpcutil.CopyRepoGCSliceToIfaces(repoGCsResp),
+	)
+
+	// clubbing repo gc responses of all peers into one
+	globalRepoGC := api.GlobalRepoGC{PeerMap: make(map[string]*api.RepoGC)}
+	for i := 0; i < len(members); i++ {
```

Comment:
I asked to copy handling from `globalPinInfoCid` but you seem to have disregarded, why?

You missed ignoring auth errors as `globalPinInfoCid` does. Also why doing the loop like `for i := 0; i < len(members); i++ {` when you could `for i, resp := range repoGCResp[i] {` like the other function. It's more readable (you are processing responses not members). There is value in keeping similar things similar.

Reply:
It is now similar to `globalPinInfoCid`